content/articles/shell-script-to-demonstrate-the-se-of-shell-function-library/index.md

```diff
@@ -0,0 +1,101 @@
+### Introduction
+The Shell Function Library is a set of functions that may be called from any location in the development environment. Shell scripting becomes a little less laborious and repetitious as a result. We may access and call such functions from other files or scripts by building a shell script with some functions specified in it. It aids in the avoidance of code repetition in huge files and sophisticated scripts.
+
+In C/C++, a Shell Function Library is similar to a header file, whereas, in Python, it is similar to a module. The current script must be aware of the function's location. To be able to use the functions, we must additionally add the file's location to the Environment variables, or run the script in the terminal to temporarily load the function library into the current shell. Shell Script Functions are similar to regular shell scripts in that they only define functions.
+
+### Table of contents
+- [Why use Function Library](#why-use-function-library)
+- [Creating function library](#creating-function-library)
+- [Using Functions From Library](#using-functions-from-library)
+- [A simple Demo of using Bash script as fuction library](a-simple-demo-of-using-bash-script-as-fuction-library)
+- [Usage of Shell Function Library](#usage-of-shell-function-library)
+- [Conclusion](#conclusion)
+
+
+### Why use Function Library
+The most important reason for having a function library is for reusability. We require functions in a function library for the following reasons.
+
+1. Sometimes we require logic to address looping logic. For example, in Oracle Applications, whenever a concurrent request is sent, we usually click on the Refresh button to retrieve the most recent data about the concurrent program's status. Or, we may create a function and place it in the function library so it can be used in other scripts.

```

Comment:
A beginner may not understand what "looping logic"is, can you talk about it a little or provide a link for the reader?